feat: Activate detour flow

joshlarson opened the pull request:
https://github.com/user-attachments/assets/0d99035d-94e9-43d9-9293-a58a0405ef08

---

Asana Ticket: https://app.asana.com/0/1205385723132845/1208153509160230/f

hannahpurcell reviewed (changes requested):
Oh my, why are the commits all the same message haha!

Looking so good! One issue, though: select a duration, click Cancel, then try to activate again. The modal still shows the selected duration, but also doesn't allow the user to proceed to "next". I believe a cancel should erase the selected duration.

![Screenshot 2024-09-18 at 10 47 29 AM](https://github.com/user-attachments/assets/fdaf6bce-f32f-417d-8efd-ef85c727055e)

hannahpurcell commented:
Also, this is postponing hi-fi styling, right? Should I be leaving styling notes or save that for later?

hannahpurcell commented on assets/tests/components/detours/diversionPage.activate.test.tsx:
```diff
@@ -70,53 +71,292 @@ const diversionPageOnReviewScreen = async (
+      expect(nextButton.get()).toBeDisabled()
+
+      expect(backButton.query()).not.toBeInTheDocument()
+      // expect(activateButton.query()).not.toBeInTheDocument()
```
```suggestion
```

hannahpurcell commented on assets/tests/components/detours/diversionPage.activate.test.tsx:
```diff
@@ -70,53 +71,292 @@ const diversionPageOnReviewScreen = async (
+      expect(nextButton.get()).toBeDisabled()
+
+      expect(backButton.query()).not.toBeInTheDocument()
+      // expect(activateButton.query()).not.toBeInTheDocument()
```
Want this check to be back in place?

joshlarson replied:
Yep!

joshlarson commented:
@hannahpurcell I think I've fixed the bug you found (well, I fixed the bug that I was smashing my head into earlier - who knew that guard clauses could be so wonderful ✨ - hopefully this is the same one)

> Also, this is postponing hi-fi styling, right?

Yeah - hi-fi styling will come later, and TBH - it may come after we go live
